File: frontend/src/utils/userRole.ts
import type { UserDetails } from '../types/ProfileTypes'

type Role = 'ROLE_BUYER' | 'ROLE_MEDIATOR' | 'ROLE_AGENCY' | 'ROLE_BRAND' | 'ROLE_ADMIN'

const roleTypeMap: Record<Role, UserDetails['type']> = {
  ROLE_BRAND:    'brand',
  ROLE_AGENCY:   'agency',
  ROLE_MEDIATOR: 'mediator',
  ROLE_BUYER:    'buyer',
  ROLE_ADMIN:    'admin',
}

export function roleToType(role: Role | null | undefined): UserDetails['type'] {
  return role ? roleTypeMap[role] : 'invalid'
}

File: backend/src/main/java/com/coddicted/buzzma/identity/entity/UserRole.java
package com.coddicted.buzzma.identity.entity;

public enum UserRole {
  ROLE_BUYER,
  ROLE_MEDIATOR,
  ROLE_AGENCY,
  ROLE_BRAND,
  ROLE_ADMIN;

  public static final class Expr {
    public static final String OR = " or ";
    public static final String AND = " and ";

    public static final String BUYER = "hasRole('BUYER')";
    public static final String MEDIATOR = "hasRole('MEDIATOR')";
    public static final String AGENCY = "hasRole('AGENCY')";
    public static final String BRAND = "hasRole('BRAND')";
    public static final String ADMIN = "hasRole('ADMIN')";
  }
}
